### DISCUSSION TITLE
formState.isDirty does not get false when reverting a number to it's default value.

### DISCUSSION THREAD

**Discussion by xeeds** [react-hook-form/react-hook-form#10178] [Q&A] [answered]
Hi
When changing value for a field in a form it is expected that formState.isDirty gets true. When reverting the value to it's defaultValue it is expected that formstate.isDirty returns to false. This works perfectly well when the value for the fieldName is a string, but not when the value is a number. I wonder why it is like that or what i am doing wrong. Please check the attached codeSandbox: https://codesandbox.io/s/autumn-cherry-eqzfy3

Changing the first input field, will make isDirty = true forever, even though changing it back to it's original value. reset() is the only way to get isDirty = false. 
The behaviour is not the same when changing the value back and forth in the second input field.

```
import "./styles.css";
import { useForm } from "react-hook-form";

export default function App() {
  const {
    register,
    formState,
    formState: { isDirty },
    reset
  } = useForm({
    defaultValues: {
      testNumber: 42,
      testNumberAsString: "42"
    }
  });

  return (
    <>
      <form>
        <input type="number" {...register("testNumber")} />
        <input type="number" {...register("testNumberAsString")} />
      </form>
      <button
        disabled={!isDirty}
        onClick={() => {
          reset();
        }}
      >
        ResetForm
      </button>
      {formState.isDirty && <p>The form is dirty!</p>}
    </>
  );
}

```

**Comment by Moshyfawn** [comment, marked as answer]
As per HTML spec, you get a `string` value from the input _(`event.target.value`)_.

The `testNumberAsString` input's default value **is** a `string`, so comparing the current and the default yields `true`.

As `testNumber` input's default value is a `number` and you get a `string` from the input when you **think** you change it to the default value, it's actually 2 different values: `42` and `"42"`.

You can pass `valueAsNumber: true` to the `testNumber` input `register` function to yield the correct behavior.

```jsx
<input type="number" {...register("testNumber", { valueAsNumber: true })} />
```

**Reply by xeeds** [reply]
Awesome, thanks!!
